lib/puma/server.rb

```diff
@@ -116,12 +119,22 @@ def uncork_socket(socket)
         rescue IOError, SystemCallError
         end
       end
+
+      def closed_socket?(socket)
+        tcp_info = socket.getsockopt(Socket::SOL_TCP, Socket::TCP_INFO)
+        state = tcp_info.unpack("C".freeze)[0]
```

Comment:
Great point, thanks @brotbert. I'll wait with updating this PR as to not lose this comment for others who may want to review.